utils/warnings.ml

```diff
@@ -212,12 +212,21 @@ let current =
       error = Array.make (last_warning_number + 1) false;
     }
 
+let disabled = ref false
+
+let without_warnings f =
+  let prev = !disabled in
+  disabled := true;
+  match f () with
+  | res -> disabled := prev; res
+  | exception e -> disabled := prev; raise e
+
```

Comment:
You could use `Misc.protect_refs` here.

Reply:
I'll change it to do that. (Although it would be slightly better if we had a `protect_ref` function as well, but I don't want to add that in this PR).